scripts/o3de/o3de/register.py

```diff
@@ -472,12 +472,12 @@ def register_restricted_path(json_data: dict,
 def register_repo(json_data: dict,
                   repo_uri: str,
                   remove: bool = False) -> int:
-    if not repo_uri:
-        logger.error(f'Repo URI cannot be empty.')
+    manifest_uri = repo.get_repo_manifest_uri(repo_uri)
+
+    if not manifest_uri:
         return 1
 
-    url = f'{repo_uri}/repo.json'
-    parsed_uri = urllib.parse.urlparse(url)
+    parsed_uri = urllib.parse.urlparse(manifest_uri)
 
     if parsed_uri.scheme in ['http', 'https', 'ftp', 'ftps']:
         while repo_uri in json_data.get('repos', []):
```

Comment:
shouldn't this code that removes a repo from json_data be inside the `if remove:` block?

Reply:
I'll add a comment here, I do remember this being unclear when reading through this bit missed adding a comment at the time.